src/modules/form/components/rhf/ControlledTextInput.tsx

```diff
@@ -0,0 +1,59 @@
+import { useCallback } from 'react';
+import { Control, useController } from 'react-hook-form';
+
+import TextInput, {
+  TextInputProps,
+} from '@/components/elements/input/TextInput';
+import { RhfRules } from '@/modules/form/types';
+
+interface ControlledTextInputProps
+  extends Omit<TextInputProps, 'value' | 'onChange'> {
+  name: string;
+  control?: Control; // Optional when using FormProvider
+  rules?: RhfRules;
+}
+
+// React-Hook-Form wrapper around TextInput
+const ControlledTextInput: React.FC<ControlledTextInputProps> = ({
+  name,
+  control,
+  onBlur,
+  rules,
+  ...props
+}) => {
+  const {
+    field,
+    fieldState: { error },
+  } = useController({
+    name,
+    control,
+    shouldUnregister: true,
+    rules: {
+      required: props.required ? 'This field is required' : false,
+      ...rules,
+    },
+  });
+
+  const wrappedOnBlur = useCallback(
```

Comment:
Am I understanding this correctly?:
- the parent can pass in a custom `onBlur` callback to this component
- the RHF also needs to know when an `onBlur` event happens, so we call its onBlur callback on line 40
- a similar `wrappedOnBlur` isn't needed in the two other controlled components added in this PR, Select and Checkbox, because we don't currently have any custom `onBlur`s on those elements in our form logic (since, right now, this is used only for the label/linkId). A `wrappedOnBlur` could be added to them in the future if we did need such functionality

Reply:
Yes that's correct.  I made a few changes with https://github.com/greenriver/hmis-frontend/pull/799/commits/ed1da8ea820269e2696298e6c586734cae32f464 to clarify this:
1)`ControlledCheckbox` and `ControlledSelect` should not _accept_ onBlur props, because they are ignored (at least right now)
2) RHF onBlur wasn't being passed to ControlledSelect, I updated that